### ISSUE TITLE
podman build: reexamine default for --force-rm

### ISSUE THREAD

**Issue by edsantiago** [containers/libpod#1586]
The (undocumented) default value for `--force-rm` is currently `false`. This can lead to surprising situations:
```
# cat >Dockerfile
FROM alpine
RUN /bin/false
^D
# podman build --format=docker .
STEP 1: FROM alpine
STEP 2: RUN /bin/false
error building at step {Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:run Args:[/bin/false] Flags:[] Attrs:map[] Message:RUN /bin/false Original:RUN /bin/false}: exit status 1
```
This isn't surprising; what is unexpected is that this has leaked an invisible, inaccessible buildah container. With enough failures, this may hog resources in a way that some users cannot diagnose (e.g. if `buildah` is not installed on the system, or if the user doesn't know to run `buildah containers`).

Another surprise might be:
```
! Rats, it failed. OK, I give up, I don't need alpine any more
# podman rmi alpine
A container associated with containers/storage, i.e. via Buildah, CRI-O, etc., may be associated with this image: 196d12cf6ab1
image is in use by a container

# buildah containers                    <--- this assumes that the user is familiar with buildah
-bash: buildah: command not found
```
This is solvable with `rmi --force`, or (for more sophisticated users) `dnf install buildah; buildah containers; buildah rm etc etc`. Again, though, there doesn't seem to be any way to investigate this using only podman.

Is this intended behavior? If it is, or if it's too late to change, could it perhaps be documented better? Perhaps adding a section to `podman-build(1)` indicating that build failures leak containers and how to remove those; filling in the `--force-rm` section and including its default and what that means; and perhaps even adding the above 'may be associated' error message to `podman-rmi(1)` with instructions on what it means and how to follow up.

**Comment by rhatdan** [comment]
@umohnani8 Could you look into this?

**Comment by umohnani8** [comment]
@edsantiago yeah following docker build, ``--force-rm` is set to false to allow users to diagnose the error that occurred during the build process by leaving the build container behind so they can peak into it. Since buildah creates containers using buildah for the build process these will be buildah containers. I am not sure how to go about it other than documenting it. if in future podman can see the buildah containers then this issue will be solved, but there really is no other way around it.
@rhatdan what do you recommend we do? I can fix up the podman build man page to elaborate this more. We could also set --force-rm to true for podman build only and document that.

**Comment by edsantiago** [comment]
Thanks @umohnani8 that's what I figured; I'm just not sure if that default makes sense for podman. It does for docker and buildah because it's the leftovers are easily found; in a podman-only setup, it's easy to imagine scenarios where resources continue to be silently consumed. I don't know what the proper solution is but I think it's worth a thoughtful discussion.

**Comment by rhatdan** [comment]
Yes lets turn the default to yes.

Leave the intermediate stuff to buildah and we can deal with this if someone complains.

**Comment by rhatdan** [comment]
BTW This is worse then this.  If you do layered builds with podman build, don't we leave buildah container layers around for ever that you would only see if you ran buildah. so if you have a pure podman effort building hundreds of containers, podman rm --all will not cleanup these left over containers, unless you blow away all images.

**Comment by umohnani8** [comment]
@rhatdan turning the default to true would not cause any problems for --layers. Although if the build fails, the user will not have any intermediate containers to look at. We can make this clear in the man page that if they want to investigate the cause of failure use buildah till we figure out how podman can view buildah containers.

**Comment by rhatdan** [comment]
@umohnani8 Ping me on IRC when you get in to discuss.

**Comment by edsantiago** [comment]
Ping; just checking in. Behavior still present in podman-0.10.1.3-2.gitdb08685.fc29 and I don't see anything relevant in more recent git commits.

**Comment by rhatdan** [comment]
Seems to be lost in the the waive of PRs.  Lets turn on the --force-rm by default for now.

**Comment by umohnani8** [comment]
Fixed in #1769. --force-rm is set to true by default in podman.
